chsdi/templates/htmlpopup/oev_haltestellen.mako

```diff
@@ -71,94 +71,96 @@ baseUrl = request.registry.settings['api_url']
+          }
+          var label = result[i].label;
+          if (label == null){ // some labels are null
+            label = '-';
+          }
+          if (label.length > 2){ // some labels are too long
+            label = label.substr(0, 3);
+          }
+          var time = moment(then, 'DD/MM/YYY HH:mm').format("HH:mm");
           var diff = moment(then,"DD/MM/YYYY HH:mm").diff(moment(now,"DD/MM/YYYY HH:mm"));
           var d = moment.duration(diff);
           var s = Math.floor(d.asMinutes()) + moment.utc(diff).format("[']", -1);
-          numero += '<p class="oev-info">' + result2[i].label + '</p>';
-          destination += '<p class="oev-info">' + result2[i].destinationName + '</p>';
-          departures += '<p class="oev-info">' + result2[i].time + '</p>';
-          timeDiff += '<p class="oev-info"><b>' + s +'</b></p>';
+          numero += '<p class="oev-info">' + label + '</p>';
+          destination += '<p class="oev-info">' + result[i].destinationName + '</p>';
+          departures += '<p class="oev-info ' + class_late + '">' + time + '</p>';
+          timeDiff += '<p class="oev-info ' + class_late + '"><b>' + s +'</b></p>';
+          delay += '<p class="oev-info oev-delay">' + late +'&nbsp;</p>';
         }
       };
       numeroCol.html(numero);
       destinationCol.html(destination);
       departuresCol.html(departures);
       timeDiffCol.html(timeDiff);
+      predictableDelayCol.html(delay);
     });
   };
 
-  var onSelect = function(val) {
+  var Refresh = function(val) {
     if (refresh) {
       clearInterval(refresh);
     }
     getInfos(val);
     refresh = setInterval(getInfos, 60000);
   };
+  Refresh();
 
-  $.getJSON( "${host}/stationboard/stops/${id}/destinations", function(result){
-    var selectDestination = '';
-    for (var i = 0; i < result.length; i++) {
-      if (result[i].destination == 'nodata') {
-        selectDestination += '<option value="' + result[i].destination + '">${_("ch.bav.haltestellen-oev.nodata")}</option>';
-      } else {
-        selectDestination += '<option value="' + result[i].name + '">' + result[i].name + '</option>';
-      }
-    };
-    select.append(selectDestination);
-    onSelect('all');
-  });
 });
 
 </script>
-    <br />
     <p><b>${c['attributes']['name'] or '-'}</b>, ${_('ch.bav.haltestellen-oev.next_departures')}:</p>
-    <select id="selectDestination${id}" class="form-control">
-      <option value="all">${_('ch.bav.haltestellen-oev.all_departures')}</option>
-    </select>
-    <br />
   <table>
     <tr>
         <td id="numero${id}" class="col-label"></td>
         <td id="destination${id}" class="col-destination"></td>
         <td id="departures${id}" class="col-departures"></td>
         <td id="timeDiff${id}" class="col-time-diff"></td>
+        <td id="predictableDelay${id}" class="col-delay"></td>
     </tr>
 % endif
     <tr>
       <td></td>
       <td></td>
+      <td></td>
     </tr>
   </table>
```

Comment:
Closing the table tag should be part of the `else:`  or I miss something . An the `<tr>` seems useless.